be/src/olap/rowset/segment_v2/binary_plain_page.h

```diff
@@ -254,6 +254,33 @@ class BinaryPlainPageDecoder : public PageDecoder {
         return Status::OK();
     };
 
+    Status read_by_rowids(const rowid_t* rowids, ordinal_t page_first_ordinal, size_t* n,
+                          vectorized::MutableColumnPtr& dst) override {
+        DCHECK(_parsed);
+        if (PREDICT_FALSE(*n == 0)) {
+            *n = 0;
+            return Status::OK();
+        }
+
+        auto data = _data.mutable_data();
+        auto total = *n;
+        size_t read_count = 0;
+        for (size_t i = 0; i < total; ++i) {
+            ordinal_t ord = rowids[i] - page_first_ordinal;
+            if (UNLIKELY(ord >= _num_elems)) {
+                *n = read_count;
+                return Status::OK();
+            }
+
+            const uint32_t start_offset = offset(ord);
+            dst->insert_data(data + start_offset, offset(ord + 1) - start_offset);
```

Comment:
There is performance issues.  For column string, it will call resize and memcpy too many times.